pkg/reconciler/buildrun/resources/image_processing.go

```diff
@@ -82,6 +85,27 @@ func SetupImageProcessing(taskRun *pipelineapi.TaskRun, cfg *config.Config, buil
 		stepArgs = append(stepArgs, convertMutateArgs("--label", labels)...)
 	}
 
+	// check if we need to set image timestamp
+	if imageTimestamp := getImageTimestamp(buildOutput, buildRunOutput); imageTimestamp != nil {
+		switch *imageTimestamp {
+		case "Zero":
+			stepArgs = append(stepArgs, "--image-timestamp", "0")
+
+		case "SourceTimestamp":
+			if !hasTaskSpecResult(taskRun, "shp-source-default-source-timestamp") {
+				return fmt.Errorf("cannot use SourceTimestamp setting, because there is no source timestamp available for this source")
+			}
+
+			stepArgs = append(stepArgs, "--image-timestamp-file", "$(results.shp-source-default-source-timestamp.path)")
+
+		case "BuildTimestamp":
+			stepArgs = append(stepArgs, "--image-timestamp", strconv.FormatInt(creationTimestamp.Unix(), 10))
+
+		default:
+			stepArgs = append(stepArgs, "--image-timestamp", *imageTimestamp)
```

Comment:
this gives u either the one from the build tool or the one defined as a unix epoch?

Reply:
ignore my previous comment.